inc/bu-template-tags.php

```diff
@@ -10,31 +10,34 @@
  * Override the default navigation menu with
  * BU custom version.
  *
- * @param string $nav Default HTML markup for the navigation.
- * @param array $args Default Responsive Framework nav arguments.
+ * @since 2.1.12
  */
 function responsive_primary_nav() {
 	if ( ! method_exists( 'BuAccessControlPlugin', 'is_site_403' ) || false === BuAccessControlPlugin::is_site_403() ) {
 
 		if ( function_exists( 'bu_navigation_display_primary' ) ) {
 			/**
-			 * Filters the BU navigation defaults.
+			 * Defines arguments to pass in to BU Navigation.
+			 *
+			 * Note: These arguments will override Primary Navigation defaults +
+			 * wp-admin settings defined in Appearance > Primary Navigation.
+			 *
+			 * @since 2.1.12
+			 *
+			 * @link https://github.com/bu-ist/bu-navigation
+			 * @see bu_navigation_display_primary in BU Navigation for all args.
+			 *
+			 * @param array $args Array of BU Navigation Primary Nav arguments.
 			 */
-			$args = apply_filters( 'bu_responsive_primary_nav_args', array(
-				'post_types'      => array( 'page' ),
-				'include_links'   => true,
-				'depth'           => BU_NAVIGATION_PRIMARY_DEPTH,
```

Comment:
It looks to me like the intention here is to honor these PHP constants, if they're defined in the theme. When we do this in other places, like with BU Branding, we disable the UI and add a note that the theme has set these defaults so that people know to ping us with a ticket if they need that adjusted.

I can see no one is using these constants right now, so we could remove it from here. But I'm wondering if we should make a change over in BU Navigation to disable the UI / settings screen for these options if they are set.

Let's say we do that. Does removing the calls here stop these constants from working in Responsive?

I have a feeling this is a case of people don't know they can use it. Especially in the case of mega navigation and other really tightly controlled navigation situations, having control over the number of items and depth of items can be a huge help in preventing unintended styling consequences later.

Reply:
You're correct - I've done a good amount of researching the bu-navigation repo and have found at the first point in time of the file where the `bu_nsbigation` exists, it waited for themes to potentially override the constants https://github.com/bu-ist/bu-navigation/blob/f0acd59bfdb5ff584f591a6993fe01cda36dc206/includes/library.php#L12-L19 and then used it as the defaults: https://github.com/bu-ist/bu-navigation/blob/f0acd59bfdb5ff584f591a6993fe01cda36dc206/includes/library.php#L747-L771 as they got merged with the plugin settings (which would then override the theme constants so not working quite as expected).

I could look into this further but wanted to prevent this theme from hardcoding these values as its broken right now and we don't want to lock in all users to 6 max items

Reply:
Definitely. Good catch btw!

Reply:
The Constants are set on line 45 and then `inc/library.php` is called on line 51.

So that code in `library.php` will never allow a theme to set the constants assuming that `bu-navigation.php` is called before the themes function.php file.

https://github.com/bu-ist/bu-navigation/blob/ccd069d43a7abdc777f3ec4fdb1ac8f6d655e25c/bu-navigation.php#L44-L51
```
// Primary navigation max items to display per level
define( 'BU_NAVIGATION_PRIMARY_MAX', 6 );
// Primary navigation maxium depth
define( 'BU_NAVIGATION_PRIMARY_DEPTH', 1 );
require_once BU_NAV_PLUGIN_DIR . '/includes/settings.php';
require_once BU_NAV_PLUGIN_DIR . '/includes/library.php';
```

Reply:
I think this seems like a good fix.

The BU-Navigation sets constants for the max.

By removing the items from the filter in Responsive as proposed, a child theme can easily override the settings for max by using the `bu_responsive_primary_nav_args` filter. That seems more the WordPress way as well rather than setting Constants.

If this fix allows the plugin to properly change the settings and honor them I think this is good.

We could also use this filter in responsive itself to set a different default max. A child theme could override it with a different priority.

Reply:
@ashleykolodziej Has some good thoughts about the UI, but it seems that behavior is not currently supported in the plugin.